lib/ansible/plugins/strategy/__init__.py

```diff
@@ -302,6 +308,96 @@ def add_tqm_variables(self, vars, play):
         vars['ansible_current_hosts'] = self.get_hosts_remaining(play)
         vars['ansible_failed_hosts'] = self.get_failed_hosts(play)
 
+    def _short_circuit_fork(self, host, task, task_vars, templar):
+        self._short_circuit_fork_conditonal(host, task, task_vars, templar)
+        self._short_circuit_fork_include(host, task, task_vars, templar)
+
+    def _short_circuit_fork_conditonal(self, host, task, task_vars, templar):
+        if not task.loop and not task.loop_with:
+            tr = None
+            try:
+                conditional = task.evaluate_conditional(templar, task_vars)
+            except AnsibleError as e:
+                tr = TaskResult(
+                    host=host.name,
+                    task=task._uuid,
+                    return_data={
+                        'failed': True,
+                        'msg': to_text(e),
+                        'exception': traceback.format_exc(),
+                    },
+                )
+            else:
+                if not conditional:
+                    tr = TaskResult(
+                        host=host.name,
+                        task=task._uuid,
+                        return_data={
+                            'changed': False,
+                            'skipped': True,
+                            'skip_reason': 'Conditional result was False',
+                        },
+                    )
+            if tr:
+                raise ForkShortCircuit(tr)
+
+    def _short_circuit_fork_include(self, host, task, task_vars, templar):
+        vars_copy = task_vars.copy()
+        vars_copy['ansible_search_path'] = task.get_search_path()
+        if not task.loop and not task.loop_with:
+            tr = None
+            # if this task is a TaskInclude, we just return now with a success code so the
+            # main thread can expand the task list for the given host
+            if task.action in ('include', 'include_tasks'):
+                include_args = task.args.copy()
+                include_file = include_args.pop('_raw_params', None)
+                if not include_file:
+                    tr = TaskResult(
```

Comment:
You set `tr` here but don't do anything with it. The code will just continue along and probably fail at 366 `include_file = templar.template(include_file)` because `include_file` is None.